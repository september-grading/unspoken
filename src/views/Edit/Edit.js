import React from 'react';
import { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { editPost, getPostById } from '../../services/posts';
import PostForm from '../../components/PostForm/PostForm';

export default function Edit() {
  const [post, setPost] = useState({});
  const [loading, setLoading] = useState(true);
  const params = useParams();

  useEffect(() => {
    const fetchData = async () => {
      const data = await getPostById(params.id);
      setPost(data);
      setLoading(false);
    };
    fetchData();
  }, [params.id, loading]);

  if (loading) {
    return <h3> Loading</h3>;
  }

  const updatePost = (key, value) => {
    post[key] = value;
    setPost({ ...post });
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      await editPost(post);
      alert('Success');
    } catch {
      alert('Error');
    }
  };

  return (
    <div>
      <PostForm post={{ ...post }} handleSubmit={handleSubmit} updatePost={updatePost} />
    </div>
  );
}
