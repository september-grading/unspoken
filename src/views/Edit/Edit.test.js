import { render, screen } from '@testing-library/react';
import Edit from '../Edit/Edit';
import { MemoryRouter, Route } from 'react-router-dom';

test('should render details and edit', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/posts/37/edit']}>
      <Route exact path="/posts/:id/edit" component={Edit} />
    </MemoryRouter>
  );
  await screen.findByDisplayValue('cuties');
  expect(container).toMatchSnapshot();
});
