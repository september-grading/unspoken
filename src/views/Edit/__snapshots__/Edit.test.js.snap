// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render details and edit 1`] = `
<div>
  <div>
    <form
      class="post-form"
    >
      <h1>
        create post
      </h1>
      <input
        name="title"
        placeholder="post title..."
        type="text"
        value="Tokio and Nori"
      />
      <textarea
        name="content"
        placeholder="tell us more... (minimum 6 characters)"
        type="text"
      >
        cuties
      </textarea>
      <input
        type="submit"
      />
    </form>
    )
  </div>
</div>
`;
